File: LastLogged/Models/TrackerCategory.swift
import Foundation
import SwiftData

@Model
final class TrackerCategory {
    @Attribute(.unique) var id: UUID
    var name: String
    var iconName: String
    var colorHex: String
    var sortOrder: Int
    var isDefault: Bool

    init(
        id: UUID = UUID(),
        name: String,
        iconName: String,
        colorHex: String,
        sortOrder: Int = 0,
        isDefault: Bool = false
    ) {
        self.id = id
        self.name = name
        self.iconName = iconName
        self.colorHex = colorHex
        self.sortOrder = sortOrder
        self.isDefault = isDefault
    }
}

File: android/app/src/main/java/com/pearsonmedia/lastlogged/data/local/entity/TrackerCategory.kt
package com.pearsonmedia.lastlogged.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "tracker_categories")
data class TrackerCategory(
    @PrimaryKey
    val id: String,

    val name: String,

    @ColumnInfo(name = "icon_name")
    val iconName: String = "category",

    @ColumnInfo(name = "color_hex")
    val colorHex: String = "#4f46e5",

    @ColumnInfo(name = "sort_order")
    val sortOrder: Int = 0,

    @ColumnInfo(name = "is_default")
    val isDefault: Boolean = false
)
